go/ops.go

```diff
@@ -20,38 +20,56 @@ import (
 	_ "golang.org/x/crypto/ripemd160" //nolint:staticcheck
 )
 
-// validate the IAVL Ops
-func validateIavlOps(op opType, b int) error {
+// validateIavlOps validates the prefix to ensure it begins with
+// the height, size, and version of the IAVL tree. Each varint must be a bounded value.
+// In addition, the remaining bytes are validated to ensure they correspond to the correct
+// length. The layerNum is the tree depth.
```

Comment:
added two more lines on leaf node/inner node constraints (feel free to take or leave) 

```suggestion
// validateIavlOps validates the prefix to ensure it begins with
// the height, size, and version of the IAVL tree. Each varint must be a bounded value.
// In addition, the remaining bytes are validated to ensure they correspond to the correct
// length. The layerNum is the tree depth. 
// For leaf nodes (depth=0), we ensure the op prefix is constrained to the height, size and
// version of the IAVL tree.
// For inner nodes (depth>=1), we ensure the remaining prefix length conforms to the expected
// length when applying inner nodes where child comes from either left or right (see comment inline).
```

Reply:
on reflection this reads a little weird. `layerNum` is passed as 0 when we're at the leaf, but in reality that is depth n, since depth of 0 usually means root. 

maybe we could change to:

```suggestion
// validateIavlOps validates the prefix to ensure it begins with
// the height, size, and version of the IAVL tree. Each varint must be a bounded value.
// In addition, the remaining bytes are validated to ensure they correspond to the correct
// length. The layerNum is the inverse of the tree depth, i.e. depth=0 means leaf, depth>=1 means inner node
```

Reply:
I like your second suggestion. It's concise and with the additional in-code comment, I think it's straightforward enough to reason about